e2e/upgrade_utils.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2024 NASD Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	sdkupgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	authoritytypes "github.com/noble-assets/authority/types"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+const (
+	haltHeightDelta    = 10 // will propose upgrade this many blocks in the future
+	blocksAfterUpgrade = 10
+
+	authModule = "noble13am065qmk680w86wya4u9refhnssqwcvgs0sfk"
```

Comment:
Quick heads up that you can get this from the Authority module types directly!

Reply:
Done! [`fc4f835` (#422)](https://github.com/noble-assets/noble/pull/422/commits/fc4f8357a339d4231913f026d628ecf27db7a82c)